core/stateful_precompile_test.go

```diff
@@ -1286,3 +1289,175 @@ func TestRewardManagerRun(t *testing.T) {
 		})
 	}
 }
+
+func TestWarpMessengerRun(t *testing.T) {
+	type test struct {
+		caller       common.Address
+		preCondition func(t *testing.T, state *state.StateDB)
+		input        func() []byte
+		suppliedGas  uint64
+		readOnly     bool
+		config       *precompile.WarpMessengerConfig
+
+		expectedRes []byte
+		expectedErr string
+
+		assertState func(t *testing.T, state *state.StateDB)
+	}
+
+	noRoleAddr := common.HexToAddress("0xF60C45c607D0f41687c94C314d300f483661E13a")
+	snowCtx := snow.DefaultContextTest()
+	snowCtx.ChainID = ids.GenerateTestID()
+	sk, err := bls.NewSecretKey()
+	require.NoError(t, err)
+	snowCtx.TeleporterSigner = teleporter.NewSigner(sk, snowCtx.ChainID)
+
+	sendInput := precompile.SendWarpMessageInput{
+		DestinationChainID: ids.GenerateTestID(),
+		DestinationAddress: [32]byte{},
+		Payload:            nil,
+	}
+
+	for name, test := range map[string]test{
+		"get blockchain ID succeeds": {
+			caller: noRoleAddr,
+			input: func() []byte {
+				input, err := precompile.PackGetBlockchainID()
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: precompile.GetBlockchainIDGasCost,
+			readOnly:    false,
+			expectedRes: func() []byte {
+				res, err := precompile.PackGetBlockchainIDOutput(snowCtx.ChainID)
+				require.NoError(t, err)
+				return res
+			}(),
+		},
+		"send warp message succeeds": {
+			caller: noRoleAddr,
+			input: func() []byte {
+				input, err := precompile.PackSendWarpMessage(sendInput)
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: precompile.SendWarpMessageGasCost,
+			readOnly:    false,
+			expectedRes: []byte{},
+		},
+		"get verified warp message no storage slots fails": {
+			caller: noRoleAddr,
+			input: func() []byte {
+				input, err := precompile.PackGetVerifiedWarpMessage(big.NewInt(0))
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: precompile.GetVerifiedWarpMessageGasCost,
+			readOnly:    false,
+			expectedErr: precompile.ErrMissingStorageSlots.Error(),
+		},
+		"readOnly get blockchain ID fails": {
+			caller: noRoleAddr,
+			input: func() []byte {
+				input, err := precompile.PackGetBlockchainID()
+				require.NoError(t, err)
+
+				return input
+			},
+			suppliedGas: precompile.GetBlockchainIDGasCost,
+			readOnly:    true,
+			expectedErr: vmerrs.ErrWriteProtection.Error(),
+		},
```

Comment:
Is there any reason not to allow this function to be marked as `view`? I think this should be allowed in a read only state.

Reply:
updated `getBlockchainID` and `getVerifiedWarpMessage` to be marked as view and allowed in a read only